src/corpora/auto.py

```diff
@@ -29,20 +34,41 @@ def get_auto_dataset(
     preprocessing_num_proc: int = 64,
     stride: int = -1,
     ignore_train: bool = False,
+    streaming: bool = True,
 ) -> datasets.DatasetDict:
     """Run basic tokenization and grouping to turn a Hugging Face Dataset (via `datasets`) into a torch.Dataset."""
 
     # Sanity check on input args
     stride = seq_len if stride < 0 else stride
     assert stride <= seq_len, f"Data grouping stride ({stride}) is smaller than sequence length: we are losing data."
+
     dataset = datasets.load_dataset(
-        dataset_id, name=dataset_name, cache_dir=str(paths["dataset"]), keep_in_memory=True
+        dataset_id, name=dataset_name, cache_dir=str(paths["dataset"]), keep_in_memory=not streaming,
+        streaming=streaming,
     )
 
+    # while HF datasets has a map method for both normal Dataset and streaming IterableDataset, the IterableDataset
+    # version doesn't actually work with Trainer so we have to wrap it in this StreamedDataset nonsense.
+    # The reasons are:
+    # 1. IterableDataset transforms are pickled using pickle (not dill or something reasonable), so you have to be super
```

Comment:
Can we upstream this as an issue in HF Datasets?

Reply:
yeah i will. I wanted to minimize to something that doesn't depend on mistral and haven't done it yet.